api/python/quilt3/data_transfer.py

```diff
@@ -295,20 +302,50 @@ def _download_file(ctx, src_bucket, src_key, src_version, dest_path):
 
     dest_file.parent.mkdir(parents=True, exist_ok=True)
 
+    with dest_file.open('wb', buffering=0) as f:
+        fileno = f.fileno()
+        is_regular_file = stat.S_ISREG(os.stat(fileno).st_mode)
+
+        if is_regular_file:
+            # Preallocate file.
+            if hasattr(os, 'posix_fallocate'):
+                os.posix_fallocate(fileno, 0, size)
+            else:
+                f.truncate(size)
+
     if src_version is not None:
         params.update(dict(VersionId=src_version))
-    resp = s3_client.get_object(**params)
 
-    body = resp['Body']
-    with open(dest_path, 'wb') as fd:
-        while True:
-            chunk = body.read(64 * 1024)
-            if not chunk:
-                break
-            fd.write(chunk)
-            ctx.progress(len(chunk))
+    chunk_ranges = (
+        get_chunk_ranges(size)
+        if is_regular_file and size >= s3_transfer_config.multipart_threshold and os.name != 'nt' else
```

Comment:
What is the significance of 'nt'?

Reply:
Multi-threaded downloads (supposedly) don't work on Windows - so it falls back to single-threaded.

Reply:
And it turns that it works on Windows, I had a wrong understanding of shared file usage on Windows.

Reply:
Oh nice!

(I tried it using Wine on Linux, and it worked... but didn't want to assume that it works on real Windows.)